src/screen.rs

```diff
@@ -1078,10 +1078,30 @@ impl Screen {
         self.need_clear_lines = false;
         self.need_clear_screen = false;
 
+        // Compare the number of lines in the (new) left prompt compared to the
+        // current prompt before it's dropped so that the lines can be cleared
+        // appropriately later (see #11875). Avoid unnecessary clearing by only
+        // doing this when `is_final_rendering`.
+        let mut prompt_has_fewer_lines = is_final_rendering;
+        if is_final_rendering {
+            let actual_prompt_line_breaks = self.actual_left_prompt.as_ref().map_or(1, |p| {
+                let mut result = 1;
+                if p.chars().any(|c| matches!(c, '\n' | '\x0C')) {
+                    result = cached_layouts
+                        .calc_prompt_layout(p, None, usize::MAX)
+                        .line_breaks
+                        .len();
+                }
+                result
+            });
+            prompt_has_fewer_lines |=
+                left_prompt_layout.line_breaks.len() < actual_prompt_line_breaks;
+        }
```

Comment:
this whole if-block doesn't do anything because
`prompt_has_fewer_lines` is always true at this point.
Judging by the comments, you want to initialize `prompt_has_fewer_lines` to `false`.